src/core/Ros.ts

```diff
@@ -689,9 +702,9 @@ export default class Ros extends EventEmitter<
           }
         } else {
           // lookup the name
-          let sub = false;
+          let sub: boolean | rosapi.TypeDef = false;
```

Comment:
Wouldn't it be more appropriate to use something like this for an optional type?
```suggestion
          let sub: rosapi.TypeDef | undefined;
```

Reply:
Agreed; this weird "or false" pattern in the codebase should be refactored out. I should open an issue about finding and replacing all these types.

Reply:
#1035